libvast/vast/system/partition.hpp

```diff
@@ -107,15 +107,13 @@ struct active_partition_state {
   /// The number of events in the partition.
   size_t events;
 
-  /// Actor handle of the index actor.
-  index_actor index;
-
   /// Actor handle of the filesystem actor.
   filesystem_actor filesystem;
 
   /// Promise that gets satisfied when the partition state was serialized
   /// and written to disk.
-  caf::typed_response_promise<atom::ok> persistence_promise;
+  caf::typed_response_promise<atom::ok, std::shared_ptr<partition_synopsis>>
```

Comment:
Not sure why the ok atom is still needed here, just the partition_synopsis should be enough for the reply.